pkg/controller/machine/controller.go

```diff
@@ -454,6 +456,15 @@ func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, er
 			klog.Errorf("Failed to update machine status %q: %v", machine.GetName(), err)
 			return err
 		}
+
+		// Update the metric after everything else has succeeded to prevent duplicate
+		// entries when there are failures
+		if phase != phaseDeleting {
```

Comment:
We should protect against duplicate calls to the same phase so we don't spoil this metric.

Reply:
I had considered this and thought it to be covered by L#432, we only update the metric on changes to the phase right? Do you think there's some extra nuance or case that needs to be covered here?

Reply:
Nope, I missed that line.  We should probably refactor this function to quit early instead of putting everything in this giant if statement.

One thing to consider is protecting against empty phase.

Reply:
> Nope, I missed that line. We should probably refactor this function to quit early instead of putting everything in this giant if statement.

Ack, that would be sensible! 

> One thing to consider is protecting against empty phase.

How do you mean here, as in if it were empty then skipped ahead to eg Running? Wondering when this could happen, only when the status is blanked somehow? Maybe on creation of master machines during IPI? How would you suggest we protect here? Should we only record `""` to `"provisioning"` and ignore `""` to anything else?

Reply:
Yeah, probably outside the scope of this patch set.  Perhaps we need a validatePhaseTransition() function to ensure we're doing the right thing here.

I'm confident the code will work as-is today, I'm worried about protecting against bugs in the future.  We can probably just make another card for this, I think we can ship as-is.